libc-bottom-half/headers/public/wasi/api.h

```diff
@@ -38,34 +34,38 @@ _Static_assert(_Alignof(int32_t) == 4, "non-wasi data layout");
 _Static_assert(_Alignof(uint32_t) == 4, "non-wasi data layout");
 _Static_assert(_Alignof(int64_t) == 8, "non-wasi data layout");
 _Static_assert(_Alignof(uint64_t) == 8, "non-wasi data layout");
-_Static_assert(_Alignof(void*) == 4, "non-wasi data layout");
+_Static_assert(_Alignof(void*) == sizeof(uintptr_t), "non-wasi data layout");
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#ifdef __cplusplus
+#if __cplusplus >= 201103L
+#define __WASI_NOEXCEPT noexcept
+#else
+#define __WASI_NOEXCEPT throw()
+#endif
+#else
+#define __WASI_NOEXCEPT
+#endif
+
 // TODO: Encoding this in witx.
 #define __WASI_DIRCOOKIE_START (UINT64_C(0))
 typedef __SIZE_TYPE__ __wasi_size_t;
 
-_Static_assert(sizeof(__wasi_size_t) == 4, "witx calculated size");
```

Comment:
Rather than removing these lines perhaps change the `4` to something like `WASM_PTR_SIZE`?

Reply:
Well the Rust generated script does not work here.

Also, personally, i would like to remove all these _Static_assert because C and C++ are designed in this way to ensure maximum portability. Better not assume anything.
In fact, in C and C++, intxx_t, uintxx_t, uintptr_t do not necessarily exist either, the standard wants you to use uint_leastxx_t.

Reply:
`
                        ret.push_str(&format!(
                            "_Static_assert(sizeof(__wasi_{}_t) == {}, \"witx calculated size\");\n",
                            ident_name(name),
                            dest.mem_size_align().size
                        ));
                        ret.push_str(&format!(
                            "_Static_assert(_Alignof(__wasi_{}_t) == {}, \"witx calculated align\");\n",
                            ident_name(name),
                            dest.mem_size_align().align
                        ));
`

it is generated by this code, but the values are not correct for 64 bit so i have to disable all of them.